### ISSUE TITLE
replace color_space field in `ImageXObject` with `color_type` field

### ISSUE THREAD

**Issue by anhtumai** [sharazam/printpdf#120]
ColorSpace is just a simplified version of `ColorType` and is not used much in this library codebase.

![image](https://user-images.githubusercontent.com/32799668/167312533-fc0eb6ee-767e-427d-ba5c-570806c9fc9b.png)

Replacing it with the `color_type` field in ImageXObject will give library users more information, in case they want to do some transformation with the field `image_data`.

For example, some people reported that adding an image with the Alpha channel will result in a blank page (#84, #119).
This can be resolved by removing the alpha channel from ImageXObject::image_data (check this comment: https://github.com/fschutt/printpdf/issues/119#issuecomment-1120434233). However, for different `color_type`, `image_data` will have different formats. For `ColorType::Rgba8`, `image_data` will have format: `[Red1, Green1, Blue1, Alpha1, ..., RedN, Green2, Blue2; AlphaN]`. But for `ColorType::Rgba16`, the format will be `[Red1a, Red1b, Green1a, Green1b, Blue1a, Blue1b, Alpha1a, Alpha1b, ...]`. Source: https://github.com/image-rs/image/issues/1711#issuecomment-1120450753

**Comment by anhtumai** [comment]
@fschutt btw, do you plan to rewrite this library to not use lopdf? To me, lopdf lacks good documentation, which can be an obstacle for people to contribute.

**Comment by fschutt** [comment]
No, I don't think I'll switch from lopdf as it's still a pretty good crate.

The reason I wanted my own ColorType is because PDF doesn't support all color types that the image crate supports, the image crate also changes a lot and I don't want to break the API.

**Comment by fschutt** [comment]
I think this can be closed, not relevant anymore. The API has changed to `RawImage::from_bytes([u8])`, it will auto-detect the color type and image type based on the crates enabled features (i.e. `png`, `jpeg`, `bmp` and so on). So this isn't relevant anymore.
